### PR TITLE
Add example for gatsby-browser.js API shouldUpdateScroll

### PR DESCRIPTION
**Pull request by amslezak**

### CODE REVIEW

File: packages/gatsby/src/utils/api-browser-docs.js [hunk 1]
```diff
@@ -35,6 +35,11 @@ exports.onRouteUpdate = true
  * @param {object} $0
  * @param {object} $0.prevRouterProps The previous state of the router before the route change.
  * @param {object} $0.pathname The new pathname
+ * @example
+ * exports.shouldUpdateScroll = (prevRouterProps, pathname) => {
+ *   window.scrollTo(0, 0)  // upon navigation, scroll to top of page
```
Comment (KyleAMathews) [inline packages/gatsby/src/utils/api-browser-docs.js]: I'm curious what the use case for doing this would be?

File: README.md [hunk 2]
```diff
@@ -152,6 +152,7 @@ Websites built with Gatsby:
 * [ERC dEX](https://ercdex.com)
 * [iContract Blog](https://blog.icontract.co.uk)
 * [Dona Rita](https://donarita.co.uk) ([source](https://github.com/peduarte/dona-rita-website))
+* [Andy Slezak](https://www.aslezak.com) ([source](https://github.com/amslezak))
```
Comment (KyleAMathews) [inline README.md]: Could you do this in a separate PR? Thanks!

File: packages/gatsby/src/utils/api-browser-docs.js (hunk 1 above)
Comment (amslezak) [reply packages/gatsby/src/utils/api-browser-docs.js]: For my particular use-case, I wanted it so when navlinks clicked always start at the top of the page.  By default, they start off at the last scrolled position of the page.  This example overrides that behavior.

**Comment by amslezak** [comment]
closing this issue, and opening separate PRs.

File: packages/gatsby/src/utils/api-browser-docs.js (hunk 1 above)
Comment (KyleAMathews) [reply packages/gatsby/src/utils/api-browser-docs.js]: Could you try upgrading to the latest Gatsby? There was a bug that was causing problems with scrolling that I fixed in https://github.com/gatsbyjs/gatsby/pull/3775

It sounds like you were just running into that bug.

I'd love an example for this API but it should be a realistic one — not one that works around a bug :-)

Comment (amslezak) [reply packages/gatsby/src/utils/api-browser-docs.js]: Understood!  I'll remove my workaround, and then test.  If I come up with a better example, I will absolutely submit.  Thanks for the update, Kyle.

Comment (amslezak) [reply packages/gatsby/src/utils/api-browser-docs.js]: Just confirming you fixed the issue I was having with your patch.
